### PR TITLE
Webapp initial terraform implementation

### PR DESCRIPTION
**Pull request by eriksendc**
Summary
---------

This PR adds terraform to create the webapp s3 bucket and populate it with a build of the web app.

Test Plan
---------

1. From the `ThreatExchange/terraform` directory, execute `terraform apply`
2. Go to the console and open the <prefix>-webapp bucket, select Properties, scroll down to the bottom section with the subheading Static website hosting, then select the bucket website endpoint.

A new tab should open showing this:
image

### CODE REVIEW

File: hasher-matcher-actioner/terraform/webapp/main.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+  provisioner "local-exec" {
+    command = "npm run build"
+    working_dir = "../webapp"
+  }
+  provisioner "local-exec" {
+    command = "aws s3 sync ../webapp/build s3://brianeriksen-webapp --acl public-read"
```
[the remark this answers is not in the document]
Comment (jeberl) [reply hasher-matcher-actioner/terraform/webapp/main.tf:32]: unsure of this is what you're saying @schatten but should definitely replace `s3://brianeriksen-webapp` with `s3://${var.prefix}-webapp`

File: hasher-matcher-actioner/webapp/.gitignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
```
Comment (jeberl) [inline hasher-matcher-actioner/webapp/.gitignore]: Why don't we want to include tests in git?

File: hasher-matcher-actioner/webapp/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
+
+Runs the app in the development mode.\
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+
+The page will reload if you make edits.\
```
Comment (jeberl) [inline hasher-matcher-actioner/webapp/README.md]: Thats cool

**Review by jeberl — APPROVED** [review]
Do it!

File: hasher-matcher-actioner/terraform/webapp/main.tf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  # For development, this makes cleanup easier
+  # If deploying for real, this should not be used
+  # Could also be set with a variable
+  force_destroy = true
```
[the remark this answers is not in the document]
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/webapp/main.tf:37]: @schatten I borrowed this from `/hashing-data/main.tf` line 28-31. I assumed this was allowing a non-empty bucket to be torn down during terraform destroy. The API ref isn't super clear, but on another page I found a comment in a script:

````# Allow deletion of non-empty bucket
````force_destroy = true

Like the comment says, I think we could / should eventually variablize this. Maybe the default should be `false` but our local copies can have `true`? What do you think?

**Pull request merged**